Added etcd to make update

jshaofuturewei opened the pull request:
Added etcd to make update
Improved etcd version comparision

Sindica commented on hack/update-etcd.sh:
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
```
Can the file be changed to executable?

Sindica commented:
I ran a test. When I have etcd linked to 3.4.4, it said does not need update.

jshaofuturewei commented:
Make update?


This e-mail and any attachments may contain confidential information from Futurewei/Huawei, which are intended only for the person or entity whose email address appears above. Any use of the information attached or contained herein in any way (including, but not limited to, total or partial disclosure, reproduction, or any unapproved dissemination) by persons other than the intended recipient(s) is prohibited. If you have received this e-mail in error, please notify the sender by phone or response email immediately and delete this original message.)


________________________________
From: Ying Huang <notifications@github.com>
Sent: Tuesday, June 23, 2020 6:06 PM
To: futurewei-cloud/arktos <arktos@noreply.github.com>
Cc: Joe Shao <joe.shao@futurewei.com>; State change <state_change@noreply.github.com>
Subject: Re: [futurewei-cloud/arktos] Added etcd to make update (#407)


I ran a test. When I have etcd linked to 3.4.4, it said does not need update.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffuturewei-cloud%2Farktos%2Fpull%2F407%23issuecomment-648522275&data=02%7C01%7Cjoe.shao%40futurewei.com%7C6d75d539b2bf4463ddd408d817dae0b0%7C0fee8ff2a3b240189c753a1d5591fedc%7C1%7C0%7C637285576129878577&sdata=jMJY7zWeJEhIfqghr7pCQ6jC1T7cK%2FJECKXSGahqE%2F8%3D&reserved=0>, or unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMD4LMU5ZO2Q2PR63MHXBCDRYFGSVANCNFSM4OF2VYGA&data=02%7C01%7Cjoe.shao%40futurewei.com%7C6d75d539b2bf4463ddd408d817dae0b0%7C0fee8ff2a3b240189c753a1d5591fedc%7C1%7C0%7C637285576129888571&sdata=jg0svMsyoKhOwxCEFOsfqbxMgllIlzNTORMsulWB1n0%3D&reserved=0>.

jshaofuturewei commented:
https://jenkins.alkaidcloud.io/job/arktos-pr-tests/1562/console

[cid:035cfd2c-e2b3-4800-88f3-e3c942aec303]




This e-mail and any attachments may contain confidential information from Futurewei/Huawei, which are intended only for the person or entity whose email address appears above. Any use of the information attached or contained herein in any way (including, but not limited to, total or partial disclosure, reproduction, or any unapproved dissemination) by persons other than the intended recipient(s) is prohibited. If you have received this e-mail in error, please notify the sender by phone or response email immediately and delete this original message.)


________________________________
From: Joe Shao <joe.shao@futurewei.com>
Sent: Tuesday, June 23, 2020 6:08 PM
To: futurewei-cloud/arktos <reply@reply.github.com>
Subject: Re: [futurewei-cloud/arktos] Added etcd to make update (#407)

Make update?


This e-mail and any attachments may contain confidential information from Futurewei/Huawei, which are intended only for the person or entity whose email address appears above. Any use of the information attached or contained herein in any way (including, but not limited to, total or partial disclosure, reproduction, or any unapproved dissemination) by persons other than the intended recipient(s) is prohibited. If you have received this e-mail in error, please notify the sender by phone or response email immediately and delete this original message.)


________________________________
From: Ying Huang <notifications@github.com>
Sent: Tuesday, June 23, 2020 6:06 PM
To: futurewei-cloud/arktos <arktos@noreply.github.com>
Cc: Joe Shao <joe.shao@futurewei.com>; State change <state_change@noreply.github.com>
Subject: Re: [futurewei-cloud/arktos] Added etcd to make update (#407)


I ran a test. When I have etcd linked to 3.4.4, it said does not need update.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffuturewei-cloud%2Farktos%2Fpull%2F407%23issuecomment-648522275&data=02%7C01%7Cjoe.shao%40futurewei.com%7C6d75d539b2bf4463ddd408d817dae0b0%7C0fee8ff2a3b240189c753a1d5591fedc%7C1%7C0%7C637285576129878577&sdata=jMJY7zWeJEhIfqghr7pCQ6jC1T7cK%2FJECKXSGahqE%2F8%3D&reserved=0>, or unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMD4LMU5ZO2Q2PR63MHXBCDRYFGSVANCNFSM4OF2VYGA&data=02%7C01%7Cjoe.shao%40futurewei.com%7C6d75d539b2bf4463ddd408d817dae0b0%7C0fee8ff2a3b240189c753a1d5591fedc%7C1%7C0%7C637285576129888571&sdata=jg0svMsyoKhOwxCEFOsfqbxMgllIlzNTORMsulWB1n0%3D&reserved=0>.

Sindica commented:
> Make update? 

I changed update-etcd.sh to be executable. And added some messages into update-etcd.sh & etcd.sh.
Ran update-etcd.sh only

Sindica commented:
> > Make update?
> 
> I changed update-etcd.sh to be executable. And added some messages into update-etcd.sh & etcd.sh.
> Ran update-etcd.sh only

ETCD_VERSION=3.4.4-arktos.1
version=3.4.4
etcd does not need update

jshaofuturewei commented:
> > > Make update?
> > 
> > 
> > I changed update-etcd.sh to be executable. And added some messages into update-etcd.sh & etcd.sh.
> > Ran update-etcd.sh only
> 
> ETCD_VERSION=3.4.4-arktos.1
> version=3.4.4
> etcd does not need update

Please check etcd by which etcd.

It might be linked to third_party/etcd


ubuntu@ip-172-30-0-240:~/go/src/arktos$ SLIENT=false hack/update-etcd.sh

etcd version 3.4.4-arktos.1 or greater required.

You can use 'hack/install-etcd.sh' to install a copy in third_party/.

The current installed etcd version is wrong. Exited.

ubuntu@ip-172-30-0-240:~/go/src/arktos$ ls third_party/ -l
total 32
-rw-rw-r-- 1 ubuntu ubuntu  896 Jun 12 15:32 BUILD
lrwxrwxrwx 1 ubuntu ubuntu   23 Jun 23 22:10 etcd -> etcd-v3.4.4-linux-amd64
-rw-rw-r-- 1 ubuntu ubuntu   24 Jun  9 02:20 etcd.BUILD
drwxr-xr-x 3 ubuntu ubuntu 4096 Feb 24 21:21 etcd-v3.4.4-linux-amd64
drwxrwxr-x 7 ubuntu ubuntu 4096 Jun  9 02:20 forked
drwxrwxr-x 2 ubuntu ubuntu 4096 Jun  9 02:20 go-srcimporter
drwxrwxr-x 2 ubuntu ubuntu 4096 Jun  9 02:20 intemp
drwxrwxr-x 3 ubuntu ubuntu 4096 Jun  9 02:20 multiarch
drwxrwxr-x 3 ubuntu ubuntu 4096 Jun  9 02:20 protobuf
ubuntu@ip-172-30-0-240:~/go/src/arktos$ etcd --version
etcd Version: 3.4.4
Git SHA: c65a9e2dd
Go Version: go1.12.12
Go OS/Arch: linux/amd64
